src/Extension/FluentExtension.php

```diff
@@ -835,4 +846,118 @@ public function cacheKeyComponent()
     {
         return 'fluentlocale-' . FluentState::singleton()->getLocale();
     }
+
+// <editor-fold defaultstate="collapsed" desc="CMS Field Augmentation">
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        // get all fields to translate and remove
+        $translated = $this->getLocalisedTables();
+        foreach ($translated as $table => $translatedFields) {
+            foreach ($translatedFields as $translatedField) {
+                // Find field matching this translated field
+                // If the translated field has an ID suffix also check for the non-suffixed version
+                // E.g. UploadField()
+                $field = $fields->dataFieldByName($translatedField);
+                if (!$field && preg_match('/^(?<field>\w+)ID$/', $translatedField, $matches)) {
+                    $field = $fields->dataFieldByName($matches['field']);
+                }
+
+                // Highlight any translatable field
+                if ($field && !$field->hasClass('LocalisedField')) {
+                    // Add a language indicator next to the fluent icon
+                    $currentLocale = Locale::getCurrentLocale();
```

Comment:
Shift this logic up above the foreach so it doesn't need rec-calculating for each field.